apps/test/unit/javalab/JavalabEditorTest.js

```diff
@@ -206,7 +208,8 @@ describe('Java Lab Editor Test', () => {
         expect(store.getState().javalab.sources[newFilename]).to.not.be
           .undefined;
         expect(store.getState().javalab.sources[oldFilename]).to.be.undefined;
-        expect(javalabEditor.state.openDialog).to.be.null;
+        expect(store.getState().javalab.editorOpenDialogName).to.be.null;
```

Comment:
Opinion I came up with when looking at these tests earlier this week: we should check the prop of the component that references this value (rather than Redux itself). Thoughts?

Reply:
Yeah, I definitely agree - I was trying to do that initially but the issue I was running into is that we don't actually read the state value (editorOpenDialogName) in the component. We only call the associated action (openEditorDialog/closeEditorDialog). I was trying to instead just assert that we called those functions, but I couldn't figure out how to do that...passing the functions as custom props didn't seem to work because we use redux inside the test so I think those functions get overwritten. Any ideas?

One option could be to remove redux entirely for this file and just assert that it calls the right action handlers. But we use redux for example to check that certain sources were updated so we'd have to change all that too.

Reply:
Gotcha, I see now. Re: Redux in tests, seems like most (all?) tests here require Redux. I feel like for a non-reusable component like this that uses Redux in our actual app, relying on Redux in tests is reasonable since that is how we actually use it? For a more reusable component, isolating tests from Redux would make more sense to me?

Reply:
Yeah I think that's fair. I think as pull more stuff out of this component into smaller sub-components, those sub-components can have more isolated unit tests (even if they're technically not reusable) and then this test could serve more as a full integration test?

Reply:
That makes sense to me, so for something like `JavalabEditorDialogManager` (if we were to add tests) we'd try to use no Redux? Do you have any rules of thumb about deciding when you might want to have a component fully isolated from Redux vs not?

Reply:
Yeah, I think that sounds right? I don't have a rule of thumb necessarily, but I think it makes sense that we test smaller, functional components without redux because that introduces a lot of overhead and leaks into areas (i.e. the Redux store) that the component doesn't need to be concerned about - as opposed a component like this (JavalabEditor) that's more complicated and needs to ensure that the UI is being updated correctly.